woocommerce/woocommerce-gutenberg-products-block
# Update Label component to TSX

The changes in this PR are:
- Add `tsconfig.json` to the `components` directory.
- Update the root `tsconfig.json` to include this project
- Rename `label/index.js` to `label/index.tsx` and convert to TypeScript.

I have not removed the JSDoc comment because I don't know if there's any alternative for typing the props we pass in.

While using JSDoc alongside TS I get this 
![image](https://user-images.githubusercontent.com/5656702/109186618-04d19100-7789-11eb-886d-1ae44ec530ff.png)
 I found https://github.com/TypeStrong/typedoc/issues/278#issuecomment-240758927 (Maybe not applicable) and https://stackoverflow.com/questions/51641378/is-there-a-way-to-document-fields-in-an-interface-in-typescript but removing the JSDoc comments and replacing them with the examples from the linked GitHub issue does not work. Neither in PHPStorm or VSCode, so maybe we want to leave the JSDoc comments in place, but maybe remove the lint that requires us to type them?

### How to test the changes in this Pull Request:

1. View a component in the browser that uses `Label` and ensure it displays properly
2. Add a `Label` component somewhere in the codebase and ensure intellisense shows the correct props and types
